Applications/Cxx/gdcmanon.cxx

```diff
@@ -173,28 +156,79 @@ static bool AnonymizeOneFile(gdcm::Anonymizer &anon, const char *filename, const
   gdcm::FileMetaInformation &fmi = file.GetHeader();
   fmi.Clear();
 
-  gdcm::Writer writer;
-  if (strcmp(outfilename, "stdout") == 0) {
-    writer.SetStream(std::cout);
-  } else {
-    writer.SetFileName(outfilename);
-  }
   writer.SetFile( file );
-  if( !writer.Write() )
-    {
-    std::cerr << "Could not Write : " << outfilename << std::endl;
-    if( strcmp(filename,outfilename) != 0 )
-      {
-      gdcm::System::RemoveFile( outfilename );
+
+  return true;
+}
+
+static bool AnonymizeOneFile(gdcm::Anonymizer &anon, const char *filename, const char *outfilename, bool continuemode) {
+    gdcm::Reader reader;
+    reader.SetFileName(filename);
+
+    if (!reader.Read()) {
+      std::cerr << "Could not read : " << filename << std::endl;
+      if (continuemode) {
+        std::cerr << "Skipping from anonymization process (continue mode)." << std::endl;
+        return true;
+      } else {
+        std::cerr << "Check [--continue] option for skipping files." << std::endl;
+        return false;
       }
-    else
-      {
-      std::cerr << "gdcmanon just corrupted: " << filename << " for you (data lost)." << std::endl;
+    }
+
+    gdcm::Writer writer;
+    writer.SetFileName(outfilename);
+    if (!AnonymizeOneFile(anon, reader, writer, filename)) {
+      return false;
+    }
+
+    if (!writer.Write()) {
+      std::cerr << "Could not Write : " << outfilename << std::endl;
+      if (strcmp(filename, outfilename) != 0) {
+        gdcm::System::RemoveFile(outfilename);
+      } else {
+        std::cerr << "gdcmanon just corrupted: " << filename << " for you (data lost)." << std::endl;
       }
+      return false;
+    }
 
-    return false;
+    return true;
+}
+
+static void anonymize_network_files(gdcm::Anonymizer &anon, unsigned short port)
+{
+  io_service ioservice;
+  tcp::endpoint endpoint(tcp::v4(), port);
+  tcp::acceptor acceptor(ioservice, endpoint);
+  std::cout << "Listening for connections on: " << port << std::endl;
+
+  while (true) {
+    tcp::iostream socket_stream;
+    boost::system::error_code connection_error;
+    acceptor.accept(*socket_stream.rdbuf(), connection_error);
+    fixed_istream_buffer seek_buffer(socket_stream, 512);
+    std::istream buffered_stream(&seek_buffer);
+
+    if (!connection_error) {
+      gdcm::Reader reader;
+      reader.SetStream(buffered_stream);
+      if(!reader.Read()) {
+        std::cerr << "Could not read tcp stream" << std::endl;
+        continue;
+      }
+
+      gdcm::Writer writer;
+      writer.SetStream(socket_stream);
+      if (AnonymizeOneFile(anon, reader, writer, "tcp stream")) {
+        socket_stream.clear();  // wont let us write due to hitting the EOF on read otherwise
+        if (!writer.Write()) {
+          std::cerr << "Could not write tcp stream" << std::endl;
+        }
+      }
+    } else {
+        std::cerr << "Could not connect: " << connection_error.message() << std::endl;
```

Comment:
The most common error is probably going to be a failure to bind the socket (`address_in_use`) because another process is using it. Like another copy of `gdcmanon` that is still running or has exited but the system hasn't fully released its socket. Should probably exit the program if this happens, it's going to be hard to properly recover from any error.

Reply:
It would be good try that actually. What happens if you start on copy of `gdcmanon`, and then try to start another on the same port?